File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.cs
public class Solution {
    public string Convert(string s, int numRows) {
        if (numRows == 1 || s.Length <= numRows) return s;
        var rows = new List<string>(new string[Math.Min(numRows, s.Length)]);
        int curRow = 0;
        bool goingDown = false;
        foreach (char c in s) {
            rows[curRow] += c;
            if (curRow == 0 || curRow == numRows - 1) goingDown = !goingDown;
            curRow += goingDown ? 1 : -1;
        }
        return string.Join("", rows);
    }
}

File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.py
class Solution(object):
    def convert(self, s, numRows):
        if numRows == 1 or len(s) <= numRows:
            return s
        rows = [''] * min(numRows, len(s))
        cur_row, going_down = 0, False
        for c in s:
            rows[cur_row] += c
            if cur_row == 0 or cur_row == numRows - 1:
                going_down = not going_down
            cur_row += 1 if going_down else -1
        return ''.join(rows)